bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java

```diff
@@ -335,21 +371,30 @@ private void refresh() throws InterruptedException, ExecutionException, TimeoutE
 
     // Help method of handleCommand to with SecondGenerationConfigurationHandler.executeConfigurationChanges method send
     // configuration changes.
-    private final void preSetExecuteConfigurationChanges(HttpClient httpClient, String url, String username,
-            String password, String dxsEntriesConf, String valueConfiguration) {
+    private final void preSetExecuteConfigurationChanges(HttpClient httpClientHandleCommand, String url,
+            String username, String password, String dxsEntriesConf, String valueConfiguration) {
         try {
-            SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
-                    dxsEntriesConf, valueConfiguration);
+            SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClientHandleCommand, url, username,
+                    password, dxsEntriesConf, valueConfiguration);
         } catch (Exception handleCommandException) {
-            logger.debug("Handle command for {} on channel {}: {}: {}: {}: {}", thing.getUID(), httpClient, url,
-                    dxsEntriesConf, valueConfiguration, handleCommandException.getMessage());
+            logger.debug("Handle command for {} on channel {}: {}: {}: {}: {}", thing.getUID(), httpClientHandleCommand,
+                    url, dxsEntriesConf, valueConfiguration, handleCommandException.getMessage());
         }
     }
 
     // Method callURL connect to inverter for value scraping
-    private final String callURL(String dxsEntriesCall)
-            throws InterruptedException, ExecutionException, TimeoutException {
-        String jsonDxsResponse = httpClient.GET(dxsEntriesCall).getContentAsString();
+    private final String callURL(String dxsEntriesCall, HttpClient httpClient) throws Exception {
+        String jsonDxsResponse = "";
+
+        try {
+            if (!httpClient.isRunning()) {
+                httpClient.start();
+            }
```

Comment:
You are using the common HTTP client. There is no reason to add this code. The HTTP client is started when your thing is initialized.